shared-helpers/src/auth/AuthContext.ts

```diff
@@ -289,33 +289,45 @@ export const AuthProvider: FunctionComponent<React.PropsWithChildren> = ({ child
         dispatch(stopLoading())
       }
     },
-    createUser: async (user: UserCreate) => {
+    createUser: async (user: UserCreate, listingIdRedirect) => {
       dispatch(startLoading())
+      const appUrl =
```

Comment:
Could you reduce the repetition of this block in these three functions? A helper function might help!